cmd/dmsgpost/dmsgpost.go

```diff
@@ -0,0 +1,226 @@
+	"github.com/skycoin/skywire-utilities/pkg/skyenv"
+	"github.com/spf13/cobra"
+
+	"github.com/skycoin/dmsg/pkg/disc"
+	dmsg "github.com/skycoin/dmsg/pkg/dmsg"
+	"github.com/skycoin/dmsg/pkg/dmsghttp"
+)
+
+var (
+	dmsgDisc     string
+	dmsgSessions int
+	dmsgpostData string
+	//	dmsgpostHeader string
+	sk            cipher.SecKey
+	dmsgpostLog   *logging.Logger
+	dmsgpostAgent string
+	logLvl        string
+)
+
+func init() {
+	rootCmd.Flags().StringVarP(&dmsgDisc, "dmsg-disc", "c", "", "dmsg discovery url default:\n"+skyenv.DmsgDiscAddr)
+	rootCmd.Flags().IntVarP(&dmsgSessions, "sess", "e", 1, "number of dmsg servers to connect to")
+	rootCmd.Flags().StringVarP(&logLvl, "loglvl", "l", "", "[ debug | warn | error | fatal | panic | trace | info ]\033[0m")
+	rootCmd.Flags().StringVarP(&dmsgpostData, "data", "d", "", "dmsghttp POST data")
+	//	rootCmd.Flags().StringVarP(&dmsgpostHeader, "header", "H", "", "Pass custom header(s) to server")
+	rootCmd.Flags().StringVarP(&dmsgpostAgent, "agent", "a", "dmsgpost/"+buildinfo.Version(), "identify as `AGENT`")
+	if os.Getenv("dmsgpost_SK") != "" {
+		sk.Set(os.Getenv("dmsgpost_SK")) //nolint
+	}
+	rootCmd.Flags().VarP(&sk, "sk", "s", "a random key is generated if unspecified\n\r")
+	var helpflag bool
+	rootCmd.SetUsageTemplate(help)
+	rootCmd.PersistentFlags().BoolVarP(&helpflag, "help", "h", false, "help for "+rootCmd.Use)
+	rootCmd.SetHelpCommand(&cobra.Command{Hidden: true})
+	rootCmd.PersistentFlags().MarkHidden("help") //nolint
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "dmsgpost",
+	Short: "dmsgpost",
+	Long: `
+	┌┬┐┌┬┐┌─┐┌─┐┌─┐┌─┐┌─┐┌┬┐
+ 	 │││││└─┐│ ┬├─┘│ │└─┐ │
+	─┴┘┴ ┴└─┘└─┘┴  └─┘└─┘ ┴ `,
+	SilenceErrors:         true,
+	SilenceUsage:          true,
+	DisableSuggestions:    true,
+	DisableFlagsInUseLine: true,
+	Version:               buildinfo.Version(),
+	PreRun: func(cmd *cobra.Command, args []string) {
+		if dmsgDisc == "" {
+			dmsgDisc = skyenv.DmsgDiscAddr
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		if dmsgpostLog == nil {
+			dmsgpostLog = logging.MustGetLogger("dmsgpost")
+		}
+		if logLvl != "" {
+			if lvl, err := logging.LevelFromString(logLvl); err == nil {
+				logging.SetLevel(lvl)
+			}
+		}
+
+		ctx, cancel := cmdutil.SignalContext(context.Background(), dmsgpostLog)
+		defer cancel()
+
+		pk, err := sk.PubKey()
+		if err != nil {
+			pk, sk = cipher.GenerateKeyPair()
+		}
+
+		u, err := parseURL(args)
+		if err != nil {
+			dmsgpostLog.WithError(err).Fatal("failed to parse provided URL")
+		}
+
+		dmsgC, closeDmsg, err := startDmsg(ctx, pk, sk)
+		if err != nil {
+			dmsgpostLog.WithError(err).Fatal("failed to start dmsg")
+		}
+		defer closeDmsg()
+
+		httpC := http.Client{Transport: dmsghttp.MakeHTTPTransport(ctx, dmsgC)}
+
+		req, err := http.NewRequest(http.MethodPost, u.URL.String(), strings.NewReader(dmsgpostData))
+		if err != nil {
+			dmsgpostLog.WithError(err).Fatal("Failed to formulate HTTP request.")
```

Comment:
Maybe `failed to build` request.

Reply:
copied from dmsgget implementation
https://github.com/skycoin/dmsg/blob/develop/cmd/dmsgget/dmsgget.go#L268